File: materials/2018-2019/11a/2019-04-22-streams/src/Person.java
import java.util.ArrayList;
import java.util.List;

public class Person {
    private String firstName;
    private String lastName;
    private int age;
    private List<Person> friends;
    private Gender gender;

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    public Person(String firstName, String lastName,
                  int age, Gender gender) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.gender = gender;

        this.friends = new ArrayList<>();
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public List<Person> getFriends() {
        return friends;
    }

    @Override
    public String toString(){
        return firstName + " " + lastName
                + " " + age + " " + gender;
    }
}

File: materials/2019-2020/consultations/2019-11-28/person/person.cc
#include "person.hh"

Person::Person(
    std::string first_name,
    std::string last_name,
    unsigned int age)
  : first_name(first_name),
    last_name(last_name),
    age(age) {}

std::string const& Person::get_first_name() const {
  return first_name;
}

std::string const& Person::get_last_name() const {
  return last_name;
}

unsigned int Person::get_age() const {
  return age;
}

std::ostream& operator<<(
    std::ostream& os,
    Person const& person) {
  return os
    << person.get_first_name() << ", "
    << person.get_last_name() << ", "
    << person.get_age();
}
